src/app/pages/OpenWalletPage/Features/FromEthereumPrivateKey/index.tsx

```diff
@@ -0,0 +1,165 @@
+import { useWalletSlice } from 'app/state/wallet'
+import { Box, Form, Heading, Paragraph, FormField, Button, TextInput, Tip } from 'grommet'
+import { View, Hide } from 'grommet-icons/icons'
+import * as React from 'react'
+import { useDispatch } from 'react-redux'
+import { OasisKey } from 'app/lib/key'
+import { base64ToUint, uint2hex } from 'app/lib/helpers'
+import { useTranslation } from 'react-i18next'
+import * as EthUtils from 'ethereumjs-util'
+import * as oasis from '@oasisprotocol/client'
+import * as oasisRT from '@oasisprotocol/client-rt'
+
+interface Props {}
+
+const parseKey = (key: string) => {
+  try {
+    const keyWithoutEnvelope = key
+      .replace(/\n/gm, '')
+      .replace(/ /g, '')
+      .replace(/^-----.*?-----/, '')
+      .replace(/-----.*?-----$/, '')
+
+    const key_bytes = base64ToUint(keyWithoutEnvelope)
+    return OasisKey.fromPrivateKey(key_bytes)
+  } catch (e) {
+    throw e
+  }
+}
+
+export async function getEvmBech32Address(evmAddress: any) {
+  if (!evmAddress) {
+    return ''
+  }
+  let newEvmAddress = evmAddress
+  if (newEvmAddress.indexOf('0x') === 0) {
+    newEvmAddress = newEvmAddress.substr(2)
+  }
+  const evmBytes = oasis.misc.fromHex(newEvmAddress)
```

Comment:
`oasis.misc.fromHex(evmAddress.replace('0x', ''))`

This also gets rid of the odd name "newEvmAddress"

Reply:
FromEthereumPrivateKey has only view logic now